### PR TITLE
Create buttons for submit and navigation

### PR DESCRIPTION
**Pull request by derjubu**
Part of #8 

Create styled components for the buttons

### CODE REVIEW

File: src/app/GlobalStyles.tsx [hunk 1]
```diff
@@ -6,16 +6,19 @@ const GlobalStyle = createGlobalStyle`
 }
 
 :root {
-  --background-title: #000fff;
+  --background-primary: #000fff;
+  --background-secondary: white;
+  --font-primary: white;
```
Comment (lajanzen) [inline src/app/GlobalStyles.tsx]:
```suggestion
  --background-secondary: #fff;
  --font-primary: #fff;
```

File: src/app/components/CardTitle/CardTitle.tsx [hunk 2]
```diff
@@ -16,9 +16,9 @@ const Cardtitle = styled.h2`
...
   margin: 0;
-  font-size: 1.5rem;
-  color: white;
-  background-color: var(--background-title);
+  font-size: 2rem;
+  color: var(--font-primary);
+  background-color: var(--background-primary);
   margin-bottom: 0.5rem;
   padding-top: 0.5rem;
   padding-bottom: 0.5rem;
```
Comment (lajanzen) [inline src/app/components/CardTitle/CardTitle.tsx]: It doesn't need to be changed but I personally prefer this:
```suggestion
  padding: 0.5rem 0;
```

File: src/app/components/NavigationButton/NavigationButton.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const Button = styled.span`
+  border: 2px #000fff solid;
+  border-radius: 5px;
+  background: var(--background-secondary);
+  color: var(--font-secondary);
+  font-size: 1.5rem;
+  font-weight: bold;
+  margin-top: 1rem;
+  margin-left: auto;
+  margin-right: auto;
```
Comment (lajanzen) [inline src/app/components/NavigationButton/NavigationButton.tsx]:
```suggestion
  margin-top: 1rem auto 0;
```

File: src/app/components/SubmitButton/SubmitButton.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  border: 2px #000fff solid;
+  border-radius: 5px;
+  background: var(--background-secondary);
+  color: var(--font-secondary);
+  font-size: 1.5rem;
+  font-weight: bold;
+  text-align: center;
+  justify-self: center;
+  grid-column: 3/5;
+  margin: 1rem 0 0.5rem 0;
```
Comment (lajanzen) [inline src/app/components/SubmitButton/SubmitButton.tsx]:
```suggestion
  margin: 1rem 0 0.5rem;
```

**Review by lajanzen — COMMENTED** [review]
🚀

File: src/app/GlobalStyles.tsx (hunk 1 above)
Comment (lmachens) [inline src/app/GlobalStyles.tsx]: I would add `color` to make sure, these variables only contain colors.
A `background` could be `linear-gradient` or `url` too.
`font` could be `font-family`.

```suggestion
  --color-background-primary: #000fff;
  --color-background-secondary: white;
  --color-primary: white;
```

File: src/app/components/NavigationButton/NavigationButton.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  onClick?: () => void;
+  children: ReactNode;
+};
+
+export default function NavigationButton({
+  children,
+  onClick,
+}: NavigationButtonProps): JSX.Element {
+  return (
+    <div>
```
Comment (lmachens) [inline src/app/components/NavigationButton/NavigationButton.tsx]:
```suggestion
```

The `<div>` should not be required.

File: src/app/components/NavigationButton/NavigationButton.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      <Button onClick={onClick}>{children}</Button>
+    </div>
+  );
+}
+
+const Button = styled.span`
```
Comment (lmachens) [inline src/app/components/NavigationButton/NavigationButton.tsx]: This `Button` is actually a `span` Element. It's confusing :)

File: src/app/components/NavigationButton/NavigationButton.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  color: var(--font-secondary);
+  font-size: 1.5rem;
+  font-weight: bold;
+  margin-top: 1rem;
+  margin-left: auto;
+  margin-right: auto;
+  padding: 0 1rem;
+  display: block;
+  width: fit-content;
+`;
```
Comment (lmachens) [inline src/app/components/NavigationButton/NavigationButton.tsx]:
```suggestion
import styled from 'styled-components';

const Button = styled.button`
  border: 2px #000fff solid;
  border-radius: 5px;
  background: var(--background-secondary);
  color: var(--font-secondary);
  font-size: 1.5rem;
  font-weight: bold;
  margin-top: 1rem;
  margin-left: auto;
  margin-right: auto;
  padding: 0 1rem;
  display: block;
  width: fit-content;
`;

export default Button ;
```

File: src/app/components/SubmitButton/SubmitButton.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+type ButtonProps = {
+  children: ReactNode;
+  onClick?: () => void;
+};
+export default function SubmitButton({
+  children,
+  onClick,
+}: ButtonProps): JSX.Element {
+  return <Button onClick={onClick}>{children}</Button>;
```
Comment (lmachens) [inline src/app/components/SubmitButton/SubmitButton.tsx]: I would expect that a `SubmitButton` has always `type="submit"`
```suggestion
  return <Button type="submit" onClick={onClick}>{children}</Button>;
```

File: src/app/components/SubmitButton/SubmitButton.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+const Button = styled.button`
+  border: 2px #000fff solid;
+  border-radius: 5px;
+  background: var(--background-secondary);
+  color: var(--font-secondary);
+  font-size: 1.5rem;
+  font-weight: bold;
+  text-align: center;
+  justify-self: center;
+  grid-column: 3/5;
```
Comment (lmachens) [inline src/app/components/SubmitButton/SubmitButton.tsx]: This is very specific and should not be in a component

```suggestion
```

File: src/app/pages/AddDestination/AddDestination.tsx [hunk 10]
```diff
@@ -63,7 +64,7 @@ export default function AddDestination(): JSX.Element {
           onChange={(event) => setEndDate(event.target.value)}
         />
       </EndTrip>
-      <button>Go!</button>
+      <SubmitButton children="Go" />
```
Comment (lmachens) [inline src/app/pages/AddDestination/AddDestination.tsx]: If you want to style this `SubmitButton` like ` grid-column: 3/5;`, you can create a styled version:

Solution 1:

```tsx
  const StyledSubmitButton = styled(SubmitButton)`
    grid-column: 3/5;
  `;
```

```tsx
<StyledSubmitButton>Go</StyledSubmitButton/>
```

Solution 2: (I prefer this)

```tsx
const Destination = styled.form`
  list-style: none;
  max-width: 1400;
  margin-left: auto;
  margin-right: auto;
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  column-gap: 5px;
  border: black solid 1px;
  border-radius: 5px;
  justify-items: center;
  text-align: center;
  
  ${SubmitButton} {
    grid-column: 3/5;
  }
`;
```

File: src/app/pages/DestinationDetailView/DestinationDetailView.tsx [hunk 11]
```diff
@@ -32,7 +32,7 @@ export default function DestinationDetailView(): JSX.Element {
           endDate: currentDestination.endDate?.replaceAll('"', ''),
         }}
       />
-      <Button onClick={goToItinerary}>Go on</Button>
+      <NavigationButton onClick={goToItinerary}>Go on</NavigationButton>
```
Comment (lmachens) [inline src/app/pages/DestinationDetailView/DestinationDetailView.tsx]: If this is only for navigation, I would use `a`. See `a vs button`.

File: src/app/pages/Itinerary/Itinerary.tsx [hunk 12]
```diff
@@ -28,7 +28,7 @@ export default function Itinerary(): JSX.Element {
           }}
         />
       ))}
-      <Button onClick={addDestination}>Add Destination</Button>
+      <NavigationButton onClick={addDestination} children="Add Destination" />
```
Comment (lmachens) [inline src/app/pages/Itinerary/Itinerary.tsx]:
```suggestion
      <NavigationButton onClick={addDestination}>Add Destination</NavigationButton>
```

**Review by lmachens — CHANGES_REQUESTED** [review]
o(*^＠^*)o

**Pull request merged**